### PR TITLE
first trials for to-one mapping support

### PR DESCRIPTION
**Pull request by koentsje**

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationHelper.java [hunk 1]
```diff
@@ -580,4 +581,26 @@ public CascadeStyle getCascadeStyle() {
...
+			PropertyAccess propertyAccess,
+			CascadeStyle cascadeStyle,
+			MappingModelCreationProcess creationProcess) {
+
+		return new SingularAssociationAttributeMapping(
+				attrName,
+				stateArrayPosition,
+				null,
+				null,
+				declaringType,
```
Comment (koentsje) [inline hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationHelper.java]: AFAICS here I would need to pass the MappingType that is determined from the EntityType attrType?

Comment (sebersole) [reply hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationHelper.java]: `EntityPersister` (which implements `EntityMappingType`) would be the `MappingType` here.  See `org.hibernate.type.EntityType#getAssociatedEntityName()`

**Comment by sebersole** [comment]
Added `org.hibernate.metamodel.mapping.internal.MappingModelCreationHelper#interpretKeyDescriptor(org.hibernate.mapping.Property, org.hibernate.mapping.ToOne, org.hibernate.persister.entity.EntityPersister, org.hibernate.dialect.Dialect, org.hibernate.metamodel.mapping.internal.MappingModelCreationProcess)` to generate this for to-ones

**Comment by dreab8** [comment]
merged

**Pull request closed without merging**